runtime/sema/type.go

```diff
@@ -9499,10 +9553,57 @@ func (t *EntitlementMapType) CheckInstantiated(_ ast.HasPosition, _ common.Memor
 	// NO-OP
 }
 
+func extractNativeTypes(
+	types []Type,
+) {
+	for len(types) > 0 {
+		lastIndex := len(types) - 1
+		curType := types[lastIndex]
+		types[lastIndex] = nil
+		types = types[:lastIndex]
+
+		switch actualType := curType.(type) {
+		case *CompositeType:
+			NativeCompositeTypes[actualType.QualifiedIdentifier()] = actualType
+
+			nestedTypes := actualType.NestedTypes
+			if nestedTypes == nil {
+				continue
+			}
+
+			nestedTypes.Foreach(func(_ string, nestedType Type) {
+				nestedCompositeType, ok := nestedType.(*CompositeType)
+				if !ok {
+					return
+				}
+
+				types = append(types, nestedCompositeType)
```

Comment:
Add all nested types (e.g. interface types), not just composite types

```suggestion
				types = append(types, nestedType)
```